### PR TITLE
Release/1.2.15

### PR DESCRIPTION
**Pull request by ikbalkaya**
Release 1.2.15

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
 # Change Log
 
+## [1.2.15](https://github.com/ably/ably-java/tree/1.2.15)
```
Comment (QuintinWillison) [inline CHANGELOG.md:3]: Version tags in this repository all have the `v` prefix:

```suggestion
## [1.2.15](https://github.com/ably/ably-java/tree/v1.2.15)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+
+[Full Changelog](https://github.com/ably/ably-java/compare/v1.2.14...1.2.15)
```
Comment (QuintinWillison) [inline CHANGELOG.md:5]:
```suggestion
[Full Changelog](https://github.com/ably/ably-java/compare/v1.2.14...v1.2.15)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,41 @@
...
+
+In this release we have added a new method that provides a completion handler for renewing an authentication token. 
+We also updated documentation to clarify thread policy for public method callbacks.
```
Comment (QuintinWillison) [inline CHANGELOG.md:8]:
```suggestion
We also updated the documentation to clarify the thread policy for public method callbacks. 
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,41 @@
 # Change Log
 
+## [1.2.15](https://github.com/ably/ably-java/tree/1.2.15)
+
+[Full Changelog](https://github.com/ably/ably-java/compare/v1.2.14...1.2.15)
+
+In this release we have added a new method that provides a completion handler for renewing an authentication token. 
+We also updated documentation to clarify thread policy for public method callbacks. 
+
+- A new `renewAuth` method was added to `Auth` and `renew` method was deprecated
```
Comment (QuintinWillison) [inline CHANGELOG.md:10]:
```suggestion
- A new `renewAuth` method was added to `Auth` and the `renew` method was deprecated
```

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Some minor observations within my review, but a much larger one is that it looks like closed issues are included that aren't yet in this release (being `HEAD` or `main` branch for our `1.2.x` series of this SDK). 🤔

**Review by KacperKluka — CHANGES_REQUESTED** [review]
The changelog contains issues from `2.0.0` which is not yet released :wink:

**Comment by ikbalkaya** [comment]
> Some minor observations within my review, but a much larger one is that it looks like closed issues are included that aren't yet in this release (being `HEAD` of `main` branch for our `1.2.x` series of this SDK). 🤔

According to https://github.com/github-changelog-generator/github-changelog-generator
we can exclude issues by labels but I haven't found anything that would allow us to automatically exclude milestone once. Hence I removed them manually

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,41 @@
...
+
+In this release we have added a new method that provides a completion handler for renewing an authentication token. 
+We also updated the documentation to clarify the thread policy for public method callbacks. 
+
+- A new `renewAuth` method was added to `Auth` and the `renew` method was deprecated
+
+**Implemented enhancements:**
+
+- Prepare the "lib" module configuration for publishing to Maven Central [\#772](https://github.com/ably/ably-java/issues/772)
+- Split library into core and platform modules [\#728](https://github.com/ably/ably-java/issues/728)
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply CHANGELOG.md]: I removed milestone related changes 5293ef3eabf40dd3dac3a0a0380c15210d56bddc

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,41 @@
...
+
+- Invalid method implementation in README [\#819](https://github.com/ably/ably-java/issues/819)
+- Document which thread is whole SDK or callbacks using [\#800](https://github.com/ably/ably-java/issues/800)
+- Use OIDC to publish from GitHub workflow runners to AWS S3 for `sdk.ably.com` deployments [\#786](https://github.com/ably/ably-java/issues/786)
+- Use the "java-library" plugin for ably-java [\#780](https://github.com/ably/ably-java/issues/780)
+- Improve build.gradle files configuration [\#779](https://github.com/ably/ably-java/issues/779)
+- Update dependency: Gradle and Gradle Android plugin com.android.tools.build:gradle [\#778](https://github.com/ably/ably-java/issues/778)
+- Update dependency: org.msgpack:msgpack-core [\#775](https://github.com/ably/ably-java/issues/775)
+- Update dependency: com.google.firebase:firebase-messaging [\#774](https://github.com/ably/ably-java/issues/774)
+- Replace the deprecated "maven" plugin with "maven-publish" [\#773](https://github.com/ably/ably-java/issues/773)
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply CHANGELOG.md]: 5293ef3eabf40dd3dac3a0a0380c15210d56bddc

**Pull request merged**